core/include/detray/tools/single_type_navigator.hpp

```diff
@@ -434,42 +453,47 @@ class single_type_navigator {
         // Sort distance to next & set navigation status
         if (not kernel.candidates.empty()) {
 
-            // Generally, we are on our way to some candidate
-            navigation.set_status(e_towards_object);
-            // This is only called after full (re-)evaluation
-            navigation.set_trust_level(e_full_trust);
-
             // Take the nearest candidate first
             std::sort(kernel.candidates.begin(), kernel.candidates.end());
             kernel.next = kernel.candidates.begin();
 
             // Are we still on an object from a previous navigation pass? Then
             // goto the next candidate.
-            // This also excludes adjacent portals -> we are towards obj again
+            // This also excludes adjacent portals -> we are on the next portal
             if (navigation() < navigation.tolerance()) {
                 // The object we are on
                 navigation.set_object(kernel.next->index);
                 // The next object that we want to reach
                 ++navigation.next();
+                // Direct hit
+                navigation.set_status(e_on_object);
                 // We might be wrong in this assumption, re-evaluate distance
                 // to next in the next pass
                 navigation.set_trust_level(e_high_trust);
+                // Call the inspector on this portal crossing, then go to next
+                navigation.run_inspector("Skipped direct hit: ");
             }
             // No current object
             else {
                 navigation.set_object(dindex_invalid);
             }
 
             navigation.set_dist(kernel.next->path);
-            return;
+            // Generally, we are on our way to some candidate
+            navigation.set_status(e_towards_object);
+            // This is only called after full (re-)evaluation
+            navigation.set_trust_level(e_full_trust);
```

Comment:
Can you explain the change here? It's overriding the if condition

Reply:
Whe call this only after a full reevaluation of the kernel. Even if it happens, that we are already on one object, we haven't moved after the reevaluation, so all the distances are still valid. I should probably get the high trust level out of the if clause. We are on object for a short time, so that the inspector does not miss the fact, that we have seen this surface (or mor likely adjacent portal)

Reply:
Okay makes sense